compiler/src/dmd/semantic3.d

```diff
@@ -980,10 +981,18 @@ private extern(C++) final class Semantic3Visitor : Visitor
                 sc2.flags = (sc2.flags & ~SCOPE.contract) | SCOPE.require;
 
                 // BUG: need to error if accessing out parameters
-                // BUG: need to disallow returns and throws
+                // BUG: need to disallow returns
                 // BUG: verify that all in and ref parameters are read
                 freq = freq.statementSemantic(sc2);
-                freq.blockExit(funcdecl, false);
+
+                const blockExit = freq.blockExit(funcdecl, false);
+                if (blockExit & BE.throw_)
+                {
+                    if (isnothrow)
+                        error(funcdecl.loc, "`%s`: `in` contract may throw but function is marked as `nothrow`", funcdecl.toPrettyChars());
```

Comment:
The usual convention is to  start with `deprecation`, and add an `@@@DEPRECATED_2.1xx@@@` comment - `xx` being the version that it should be made an error, take the next version and add 10.

Reply:
Thanks, I also added an according changelog entry and opened dlang/dlang.org#3392 to update the list of deprecated features.